feathr_project/feathr/registry/_feature_registry_purview.py

```diff
@@ -733,18 +735,25 @@ def _delete_all_feathr_entities(self):
 
         :param guid: The guid or guids you want to remove.
         """
-        entities = self.purview_client.discovery.search_entities(
-            "feathr*", limit=20)
+        # should not be large than this, otherwise the backend might throw out error
+        batch_delte_size = 100
 
-        # [print(entity) for entity in entities]
-        guid_list = [entity["id"] for entity in entities]
+        # use the `query` API so that it can return immediatelly (don't use the search_entity API as it will try to return all the results in a single request)
 
-        # should not be large than this, otherwise the backend might throw out error
-        batch_delte_size = 15
-        for i in range(0, len(guid_list), batch_delte_size):
-            self.purview_client.delete_entity(
-                guid=guid_list[i:i+batch_delte_size])
+        while True:
+            result = self.purview_client.discovery.query(
+                "feathr", limit=batch_delte_size)
+            logger.info("Total number of entities:",result['@search.count'] )
+
+            # if no results, break:
+            if result['@search.count']  == 0:
+                break
+            entities = result['value']
+            guid_list = [entity["id"] for entity in entities]
+            self.purview_client.delete_entity(guid=guid_list)
             logger.info("{} feathr entities deleted", batch_delte_size)
+            # sleep here, otherwise backend might throttle
+            sleep(1)
```

Comment:
This will issue delete entity rest call for every feature to be deleted, and likely to hit rate limit and get throttled.

Can be optimized by batch deletion. Ref: [https://docs.microsoft.com/en-us/rest/api/purview/catalogdataplane/entity/delete-by-guids](https://docs.microsoft.com/en-us/rest/api/purview/catalogdataplane/entity/delete-by-guids)

Reply:
Thanks... This API already delete in a batch way (the input is a guid_list). See the implementation of this API here (https://github.com/wjohnson/pyapacheatlas/blob/master/pyapacheatlas/core/client.py#L83) which is using exactly the REST endpoint you put.